### PR TITLE
Add function that reports hyperparameter metric

### PR DESCRIPTION
**Pull request by ana-simionescu**
Created a function in train_util.py that reports hyperparameter metric to AI Platform.
The function is called at each step of the training.

### CODE REVIEW

File: ddsp/training/train_util.py [hunk 1]
```diff
@@ -140,6 +140,13 @@ def format_for_tensorboard(line):
     tf.summary.text(name='gin/' + base_name, data=text_tensor, step=step)
     summary_writer.flush()
 
+def report_metric_to_ai_platform(loss_report, step):
```
Comment (jesseengel) [inline ddsp/training/train_util.py:143]: nit, I would make the loss_report arg `metric_value` and add a kwarg at the end `tag='Loss'` to make this function more general.

I would also call it `report_metric_to_hypertune()` as I think you can use the package outside ai-platform as well

File: ddsp/training/train_util.py [hunk 2]
```diff
@@ -230,6 +237,9 @@ def train(data_provider,
           tf.summary.scalar('losses/{}'.format(k), metric.result(), step=step)
           metric.reset_states()
 
+      # Report metrics for hyperparameter tuning.
```
Comment (jesseengel) [inline ddsp/training/train_util.py:240]: I would add a kwarg to `train()` called 'report_loss_to_hypertune=False` and then wrap this line in an if statement.

Then you can set that kwarg based on the flag in ddsp_run.py

File: ddsp/training/ddsp_run.py [hunk 3]
```diff
@@ -89,6 +89,9 @@
 flags.DEFINE_boolean('allow_memory_growth', False,
                      'Whether to grow the GPU memory usage as is needed by the '
                      'process. Prevents crashes on GPUs with smaller memory.')
+flags.DEFINE_boolean('hp_tuning', False,
+                     'Wheter to enable reporting for hyperparameter tuning '
+                     'on AI Platform')
```
Comment (jesseengel) [inline ddsp/training/ddsp_run.py:94]: nit, end sentence with `.`

File: ddsp/training/train_util.py [hunk 4]
```diff
@@ -230,6 +239,10 @@ def train(data_provider,
           tf.summary.scalar('losses/{}'.format(k), metric.result(), step=step)
           metric.reset_states()
 
+      # Report metrics for hyperparameter tuning if enabled
```
Comment (jesseengel) [inline ddsp/training/train_util.py:242]: nit, end sentence with `.`
